src/dtgtk/thumbtable.c

```diff
@@ -2176,6 +2176,64 @@ gboolean dt_thumbtable_ensure_imgid_visibility(dt_thumbtable_t *table, const int
   return FALSE;
 }
 
+static gboolean _filemanager_check_rowid_visibility(dt_thumbtable_t *table, const int rowid)
+{
+  if(rowid < 1) return FALSE;
+  if(!table->list || g_list_length(table->list) == 0) return FALSE;
```

Comment:
maybe you need to check the whole table->widget visibility too. Otherwise it may return TRUE in darkroom, ...
(same for zoomable)

Reply:
It's wired into the Lua API at darktable.gui.views.lighttable.is_image_visible().  I'll do some further testing and see if I what happens if I try and call it while I'm in darkroom view.